nextcord/application_command.py

```diff
@@ -0,0 +1,883 @@
+from __future__ import annotations
+import asyncio
+from inspect import signature, Parameter
+from typing import Any, Callable, Dict, List, Optional, Set, TYPE_CHECKING, Tuple, Union
+
+from . import utils
+from .abc import GuildChannel
+from .enums import ApplicationCommandType, ApplicationCommandOptionType, ChannelType
+from .interactions import Interaction
+from .member import Member
+from .message import Message
+from .mixins import Hashable
+from .role import Role
+from .user import User
+
+if TYPE_CHECKING:
+    from .guild import Guild
+    from .state import ConnectionState
+
+
+__all__ = (
+    'ApplicationCommandResponseOptionChoice',
+    'ApplicationCommandResponseOption',
+    'ApplicationCommandResponse',
+    'SlashOption',
+    'ApplicationCommand',
+    'ApplicationSubcommand',
+    'CommandOption',
+    'ClientCog',
+    'InvalidCommandType',
+    'slash_command',
+    'message_command',
+    'user_command'
+)
+
+
+class InvalidCommandType(Exception):
+    """Raised when an unhandled Application Command type is encountered."""
+    pass
+
+
+class ApplicationCommandResponse(Hashable):
+    """Represents the response that Discord sends back when queried for Application Commands.
+
+    Attributes
+    ----------
+    id: :class:`int`
+        Discord ID of the Application Command.
+    type: :class:`nextcord.ApplicationCommandType`
+        Enum corresponding to the Application Command type. (slash, message, user)
+    guild_id: Optional[:class:`int`]
+        The Guild ID associated with the Application Command. If None, it's a global command.
+    name: :class:`str`
+        Name of the Application Command.
+    description: :class:`str`
+        Description of the Application Command.
+    options: List[:class:`nextcord.ApplicationCommandResponseOption`]
+        A list of options or subcommands that the Application Command has.
+    default_permission: :class:`bool`
+        If the command is enabled for users by default.
+    """
+
+    def __init__(self, state: ConnectionState, payload: dict):
+        self._state: ConnectionState = state
+        self.id: int = int(payload["id"])
+        self.type: ApplicationCommandType = ApplicationCommandType(payload["type"])
+        self.guild_id: Optional[int] = utils._get_as_snowflake(payload, "guild_id")
+        self.name: str = payload["name"]
+        self.description: str = payload["description"]
+        self.options: List[
+            ApplicationCommandResponseOption
+        ] = ApplicationCommandResponseOption._create_options(payload.get("options", []))
+        self.default_permission: Optional[bool] = payload.get(
+            "default_permission", True
+        )
+
+    @property
+    def guild(self) -> Optional[Guild]:
+        """Optional[:class:`Guild`]: Returns the :class:`Guild` associated with this Response, if any."""
```

Comment:
Shouldnt this be something like
```py
"""
Returns
---------
Optional[:class:`Guild`]
    Returns the :class:`Guild` associated with this Response, if any.
"""

Reply:
Nah, properties can work like this to shorten strings @Skelmis